File: C#.Net/JDI/Web/JDI Web/Attributes/Objects/FillFromAnnotationRules.cs
using System.Linq;
using JDI_Web.Selenium.Elements.Complex;
using JDI_Web.Selenium.Elements.Complex.Table;
using static System.Int32;
using static Epam.JDI.Core.Settings.JDISettings;

namespace JDI_Web.Attributes.Objects
{
    public class FillFromAnnotationRules
    {
        public static void SetUpTable(Table table, JTableAttribute jTable)
        {
            table.SetUp(jTable.Root.ByLocator, jTable.Cell.ByLocator,
                    jTable.Row.ByLocator, jTable.Column.ByLocator, jTable.Footer.ByLocator,
                    jTable.ColStartIndex, jTable.RowStartIndex);

            if (jTable.Header != null)
                table.ColumnHeaders = jTable.Header;
            if (jTable.RowsHeader != null)
                table.RowHeaders = jTable.RowsHeader;

            if (jTable.Height > 0)
                table.SetColumnsCount(jTable.Height);
            if (jTable.Width > 0)
                table.SetRowsCount(jTable.Width);
            if (!jTable.Size.Equals(""))
            {
                var split = jTable.Size.Split('x');
                if (split.Length == 1)
                    split = jTable.Size.Split('X');
                if (split.Length != 2)
                    throw Exception("Can't setup Table from attribute. Bad size: " + jTable.Size);
                table.SetColumnsCount(Parse(split[0]));
                table.SetRowsCount(Parse(split[1]));
            }
            table.HeaderType = jTable.HeaderType;
            table.UseCache(jTable.UseCache);
        }

        public static void SetUpDropdown(Dropdown dropdown, JDropdownAttribute jDropdown)
        {
            dropdown.SetUp(jDropdown.Root.ByLocator, jDropdown.Value.ByLocator,
                    jDropdown.List.ByLocator, jDropdown.Expand.ByLocator,
                    jDropdown.ElementByName.ByLocator);
        }
        public static void SetUpMenu(Menu menu, JMenuAttribute jMenu)
        {
            menu.SetUp(jMenu.LevelLocators.Select(findby => findby.ByLocator).ToList());
            if (!jMenu.Separator.Equals(""))
                menu.UseSeparator(jMenu.Separator);
        }
    }
}

File: Java/JDI/jdi-uitest-mobile/src/main/java/com/epam/jdi/uitests/mobile/appium/elements/pageobjects/annotations/objects/FillFromAnnotationRules.java
package com.epam.jdi.uitests.mobile.appium.elements.pageobjects.annotations.objects;

import com.epam.jdi.uitests.mobile.appium.elements.complex.Dropdown;
import com.epam.jdi.uitests.mobile.appium.elements.complex.Menu;
import com.epam.jdi.uitests.mobile.appium.elements.complex.table.Table;
import com.epam.jdi.uitests.mobile.appium.elements.pageobjects.annotations.AppiumAnnotationsUtil;

import static com.epam.commons.LinqUtils.select;
import static com.epam.jdi.uitests.core.settings.JDISettings.exception;
import static com.epam.jdi.uitests.mobile.appium.elements.pageobjects.annotations.AppiumAnnotationsUtil.findByToBy;
import static java.lang.Integer.parseInt;
import static java.util.Arrays.asList;

/**
 * Created by 12345 on 17.05.2016.
 */
public class FillFromAnnotationRules {

    public static void setUpTable(Table table, JTable jTable) {
        table.setUp(findByToBy(jTable.root()), findByToBy(jTable.cell()),
                findByToBy(jTable.row()), findByToBy(jTable.column()), findByToBy(jTable.footer()),
                jTable.colStartIndex(), jTable.rowStartIndex());

        if (jTable.header().length > 0)
            table.hasColumnHeaders(asList(jTable.header()));
        if (jTable.rowsHeader().length > 0)
            table.hasRowHeaders(asList(jTable.rowsHeader()));

        if (jTable.height() > 0)
            table.setColumnsCount(jTable.height());
        if (jTable.width() > 0)
            table.setRowsCount(jTable.width());
        if (!jTable.size().equals("")) {
            String[] split = jTable.size().split("x");
            if (split.length == 1)
                split = jTable.size().split("X");
            if (split.length != 2)
                throw exception("Can't setup Table from attribute. Bad size: " + jTable.size());
            table.setColumnsCount(parseInt(split[0]));
            table.setRowsCount(parseInt(split[1]));
        }

        switch (jTable.headerType()) {
            case COLUMNS_HEADERS:
                table.hasOnlyColumnHeaders();
                break;
            case ROWS_HEADERS:
                table.hasOnlyRowHeaders();
                break;
            case ALL_HEADERS:
                table.hasAllHeaders();
                break;
            case NO_HEADERS:
                table.hasNoHeaders();
                break;
        }
        table.useCache(jTable.useCache());
    }

    public static void setUpDropdown(Dropdown dropdown, JDropdown jDropdown) {
        dropdown.setUp(findByToBy(jDropdown.root()), findByToBy(jDropdown.value()),
                findByToBy(jDropdown.list()), findByToBy(jDropdown.expand()),
                findByToBy(jDropdown.elementByName()));
    }
    public static void setUpMenu(Menu menu, JMenu jMenu) {
        menu.setUp(select(asList(jMenu.levelLocators()), AppiumAnnotationsUtil::findByToBy));
        if (!jMenu.separator().equals(""))
            menu.useSeparator(jMenu.separator());
    }
}
